superduper/base/datalayer.py

```diff
@@ -898,8 +900,11 @@ def _add_component_to_cache(self, component: Component):
         """
         type_id = component.type_id
         if cm := self.type_id_to_cache_mapping.get(type_id):
+            # NOTE: We need to reload the object since in `schedule_jobs`
+            # of the object, `db.replace` might be performed.
+            # e.g model prediction object is replace with updated datatype.
+            self.load(type_id, component.identifier)
```

Comment:
Should we add a reload operation in the db.replace function?

Reply:
yes, we can do it